apps/studio/src/components/AppNavbar/PublishButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { useToast } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+
+import { trpc } from "~/utils/trpc"
+
+interface PublishButtonProps {
+  pageId?: string
+  siteId?: string
+}
+
+const PublishButton = ({ pageId, siteId }: PublishButtonProps): JSX.Element => {
+  const toast = useToast()
+  const { mutate, isLoading } = trpc.page.publishPage.useMutation({
+    onSuccess: (data) => {
+      if (data.versionId) {
+        toast({
+          status: "success",
+          title: "Page published successfully",
+        })
+      } else {
+        toast({
+          status: "error",
+          title: data.error,
+        })
+      }
+    },
+    onError: () => {
+      toast({
+        status: "error",
+        title: "Failed to publish page",
+      })
+    },
+  })
+
+  const handlePublish = () => {
+    const coercedSiteId = Number(siteId)
+    const coercedPageId = Number(pageId)
+    if (coercedSiteId && coercedPageId)
+      mutate({ pageId: coercedPageId, siteId: coercedSiteId })
+  }
+  return (
+    <Button
+      variant="solid"
+      size="sm"
+      onClick={handlePublish}
+      isLoading={isLoading}
+    >
+      Publish
+    </Button>
```

Comment:
issue: I think we should either:
1. Disable this publish button if the page has never been saved before (i.e. draftBlobId is null/missing). I think in the procedure you did this, but it is not reflected in the user interface.
2. Immediately publish the unsaved page state, even if the draftBlobId is null/missing, so that this button can do something meaningful.

I'm more in favour of the 1st option because it seems safer and having no draftBlobId usually means an empty page which we would want to avoid publishing, but no strong opinion.

Reply:
Yup makes sense, have implemented option 1 as well